src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -303,17 +303,28 @@ class LedgerImpl(vm: VM, blockchain: BlockchainImpl, blockchainConfig: Blockchai
     worldStateProxy.saveAccount(senderAddress, account.increaseBalance(-calculateUpfrontGas(stx.tx)).increaseNonce)
   }
 
-  private[ledger] def prepareProgramContext(stx: SignedTransaction, blockHeader: BlockHeader, worldStateProxy: InMemoryWorldStateProxy, config: EvmConfig): PC =
-    stx.tx.receivingAddress match {
-      case None =>
-        val address = worldStateProxy.createAddress(creatorAddr = stx.senderAddress)
-        val world1 = worldStateProxy.transfer(stx.senderAddress, address, UInt256(stx.tx.value))
-        ProgramContext(stx, address,  Program(stx.tx.payload), blockHeader, world1, config)
+  private[ledger] def prepareProgramContext(stx: SignedTransaction, blockHeader: BlockHeader, worldStateProxy: InMemoryWorldStateProxy,
+                                            config: EvmConfig): PC = {
+    def prepareContext: PC = {
+      val address = worldStateProxy.createAddress(creatorAddr = stx.senderAddress)
+      val worldAfterTransfer = worldStateProxy.transfer(stx.senderAddress, address, UInt256(stx.tx.value))
+
+      worldAfterTransfer.getAccount(address) match {
+        case None => ProgramContext(stx, address,  Program(stx.tx.payload), blockHeader, worldAfterTransfer, config)
+        case Some(acc) =>
```

Comment:
So either way we create a new empty account, right? Do we then need to check for its prior existence?

Reply:
It is not strictly speaking empty (its ether balance is not cleared).
We can simplify this function to avoid pattern matching, as `transfer` already creates an account if one is not existent, but nevertheless, we need to `getAccount(address)` to clear it. 
We could theoretically make some special transfer function which would clear account during transfer, but I don't think it is necessary as this code path has almost 0 probability of execution.